### PR TITLE
Build job-scheduler dependency in CI vs. zip

### LINKED ISSUE #17: [BUG] Checked in versions of job-scheduler-*-SNAPSHOT.zip
**Is your feature request related to a problem? Please describe.**
We commit the job-scheduler depenency into src/test/resources/job-scheduler/opendistro-job-scheduler-1.15.0.0-SNAPSHOT.zip and similar which is just some random version built on a developer desktop. This is not reliable.

**Describe the solution you'd like**
Replace with a dynamic maven dependency or build one on demand in CI.

**Describe alternatives you've considered**
This is what it takes to update this today.

```
cd job-scheduler
git checkout 1.2
git pull
./gradlew assemble
rm ../anomaly-detection/src/test/resources/job-scheduler/*
rm -rf ../anomaly-detection/src/test/resources/org/opensearch/ad/bwc/job-scheduler/*
cp ./build/distributions/opensearch-job-scheduler-1.2.3.0-SNAPSHOT.zip ../anomaly-detection/src/test/resources/job-scheduler/
mkdir -p ../anomaly-detection/src/test/resources/org/opensearch/ad/bwc/job-scheduler/1.2.3.0-SNAPSHOT
cp ./build/distributions/opensearch-job-scheduler-1.2.3.0-SNAPSHOT.zip ../anomaly-detection/src/test/resources/org/opensearch/ad/bwc/job-scheduler/1.2.3.0-SNAPSHOT
```

**Additional context**
Part of #15, right now `integTest` is failing because of version incompatibility.

```
| Output for ./bin/opensearch-plugin:-> Installing file:/home/ubuntu/source/opensearch-project/anomaly-detection/dblock/src/test/resources/job-scheduler/opendistro-job-scheduler-1.15.0.0-SNAPSHOT.zip
| -> Downloading file:/home/ubuntu/source/opensearch-project/anomaly-detection/dblock/src/test/resources/job-scheduler/opendistro-job-scheduler-1.15.0.0-SNAPSHOT.zip
| -> Failed installing file:/home/ubuntu/source/opensearch-project/anomaly-detection/dblock/src/test/resources/job-scheduler/opendistro-job-scheduler-1.15.0.0-SNAPSHOT.zip
| -> Rolling back file:/home/ubuntu/source/opensearch-project/anomaly-detection/dblock/src/test/resources/job-scheduler/opendistro-job-scheduler-1.15.0.0-SNAPSHOT.zip
| -> Rolled back file:/home/ubuntu/source/opensearch-project/anomaly-detection/dblock/src/test/resources/job-scheduler/opendistro-job-scheduler-1.15.0.0-SNAPSHOT.zip
| Exception in thread "main" java.lang.IllegalArgumentException: Plugin [opendistro-job-scheduler] was built for OpenSearch version 7.10.3 but version 1.0.0 is running
|       at org.opensearch.plugins.PluginsService.verifyCompatibility(PluginsService.java:365)
|       at org.opensearch.plugins.InstallPluginCommand.loadPluginInfo(InstallPluginCommand.java:801)
|       at org.opensearch.plugins.InstallPluginCommand.installPlugin(InstallPluginCommand.java:856)
|       at org.opensearch.plugins.InstallPluginCommand.execute(InstallPluginCommand.java:263)
|       at org.opensearch.plugins.InstallPluginCommand.execute(InstallPluginCommand.java:237)
|       at org.opensearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:99)
|       at org.opensearch.cli.Command.mainWithoutErrorHandling(Command.java:140)
|       at org.opensearch.cli.MultiCommand.execute(MultiCommand.java:104)
|       at org.opensearch.cli.Command.mainWithoutErrorHandling(Command.java:140)
|       at org.opensearch.cli.Command.main(Command.java:103)
|       at org.opensearch.plugins.PluginCli.main(PluginCli.java:60)
```

### PR DESCRIPTION
**Pull request by dblock**
Signed-off-by: dblock <dblock@amazon.com>

### Description

Build and use job-scheduler artifact used in local integration testing instead of having a .zip version checked in. This avoids problems described in #26 and having to always manually add job-scheduler.zip every time we upgrade the dependency. 

I switched the order of checkout for code because every time GitHub checkout@v2 runs in a folder it runs a `git clean` that removes all un-versioned files.
 
### Issues Resolved

Closes #17. 
 
### Check List

- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -53,6 +53,12 @@ jobs:
       - name: Build job-scheduler
         working-directory: ./job-scheduler
         run: ./gradlew publishToMavenLocal -Dopensearch.version=1.0.0-alpha1 -Dbuild.snapshot=false
+      - name: Assemble job-scheduler
```
Comment (ylwu-amzn) [inline .github/workflows/CI.yml:56]: From the error of https://github.com/opensearch-project/anomaly-detection/pull/25/checks?check_run_id=2440286543
> Could not find com.amazon.opendistroforelasticsearch:opendistro-job-scheduler-spi:1.0.0.0.

We should build job scheduler SPI as well.

Comment (dblock) [reply .github/workflows/CI.yml:63]: The real issue here is that github checkout deletes everything in the current folder when checking out 

```
Deleting the contents of '/home/runner/work/anomaly-detection/anomaly-detection'
```

**Comment by dblock** [comment]
This is ready to review @ylwu-amzn!

**Review by ylwu-amzn — APPROVED** [review]
LGTM. Thanks for the change!

**Pull request merged**